File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length; 
        int[] res = new int[n]; 
        Stack<Integer> stack = new Stack<>(); 

        for(int i = n - 1; i >= 0; i--){
            while(!stack.empty() && temperatures[stack.peek()] <= temperatures[i]) stack.pop();
            if(stack.isEmpty()) res[i] = 0;  
            else res[i] = stack.peek() - i; 
            stack.push(i);
        }

        return res; 
    }
}

File: Data Structures & Algorithms/concatenation-of-array/submission-1.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)

        res = [None] * (2 *n)

        for i in range(n): 
            res[i] = nums[i]
            res[i+n] = nums[i]

        return res


        